velox/functions/prestosql/ArraySort.cpp

```diff
@@ -64,11 +69,16 @@ BufferPtr sortElements(
           if (bNull) {
             return true;
           }
-          return baseElementsVector->compare(
-                     baseElementsVector,
-                     decodedIndices[a],
-                     decodedIndices[b],
-                     flags) < 0;
+
+          std::optional<int32_t> result = baseElementsVector->compare(
+              baseElementsVector, decodedIndices[a], decodedIndices[b], flags);
+
+          VELOX_USER_CHECK(
```

Comment:
I think some more work needs to be done to report these errors properly.

See https://facebookincubator.github.io/velox/develop/scalar-functions.html#try-expression-support and https://velox-lib.io/blog/optimize-try_cast.

Reply:
@mbasmanova Learn from this article and your PR https://github.com/facebookincubator/velox/pull/6891, I've optimized this part by doing the following,
1. Early throw exception and return in the sort function of each row.
2. Set error in the context.
3. Add try(array_sort(c0)) test cases.

Reply:
Please, check out https://velox-lib.io/blog/optimize-try_cast./ as well.

Reply:
> Please, check out https://velox-lib.io/blog/optimize-try_cast./ as well.

@mbasmanova According to this blog, other than folly and cast-related optimization, it optimizes the `applyToSelectedNoThrow ` by introducing `setVeloxExceptionError` and furthermore it avoids VeloxException construction by only tracking rows that have an error using a selectivity vector for `try` only need to be converted to null if is thrown. For `array_sort` we still need to construct and VeloxException and track them using the `ErrorVector`. Do I understand correctly?

Reply:
I believe this is so. Thank you for double checking. CC: @laithsakka